Weather/Weather/WViewController/ViewController.m

```diff
@@ -0,0 +1,158 @@
+//
+//  ViewController.m
+//  Weather
+//
+//  Created by Sangwook Wi on 5/2/17.
+//  Copyright © 2017 Sangwook Wi. All rights reserved.
+//
+
+#import "ViewController.h"
+#import "WModel.h"
+#import "WCenter.h"
+#import "WTableViewCell.h"
+
+@interface ViewController() {
+    WCenter *cInstance;
+    CGRect originalsize;
+    NSArray *dayModel;
+}
+
+@property(strong, nonatomic) NSMutableArray *daysCell;
+@property(strong, nonatomic) NSMutableArray *highTempCell;
+@property(strong, nonatomic) NSMutableArray *lowTempCell;
+@property(strong, nonatomic) NSMutableArray *weatherCell;
+@property(strong, nonatomic) WModel *cityModel;
+
+@end
+
+@implementation ViewController
+
+bool locationViewOpened;
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    locationViewOpened = false;
+    [self unhide];
+    _locationTextField.returnKeyType = UIReturnKeyDone;
+    [_locationTextField setDelegate:self];
+    cInstance = [WCenter sharedInstance];
+    [cInstance fetchModelData:^(WModel *dataModel) {
+        [self update:dataModel];
```

Comment:
Interesting that you fetch your data after the view did load - looks like there would be a wait for the info to populate in, so maybe we want to consider getting the data fetch going ahead of time. Especially important to note when we're working with larger amounts of data to parse in the future.